bundles/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/internal/handler/RobonectHandler.java

```diff
@@ -71,12 +74,16 @@
     private ScheduledFuture<?> pollingJob;
 
     private HttpClient httpClient;
+    private TimeZoneProvider timeZoneProvider;
+
+    private ZoneId timeZone = ZoneId.of("Europe/Berlin");
```

Comment:
I think instead of caching this value in a field, you might consider dynamically creating it every time since its value should fall back to the `timeZoneProvider` timezone whenever it isn't specified by the config. 
The problem I see is that the value returned by `timeZoneProvider` might change over time if the user changes settings in runtime.
So any value returned by timeZoneProvider should never be cached since it might change at any time, without notification.

Reply:
My suggestion is to more or less replace uses of `timeZone` with a new method `getTimeZone()`.

Reply:
I haven't thought about this. thanks for pointing it out!

I have added a commit to change this.

Reply:
I thought about this again and noticed that if you change the configuration of the thing, the handler gets restarted anyway, because it doesn't overwrite `handleConfigurationUpdate` and thus the `BaseThingHandler#handleConfigurationUpdate` will call `dispose()` and `initialize()`, so my previous implementation would also read the changed value within `initialize()`.

I leave it up to you, to include the commit with the change or to drop it during merge.

Reply:
Edit: I decided to drop my commit because it was difficult to fix the tests (accessing the config ran into an NPE).

Reply:
I was referring to the return value of `timeZoneProvider` changing during runtime. A change in the return value wouldn't notify the handler and it wouldn't restart the handler or binding in any way. The binding shouldn't save what timeZoneProvider return into a field.

Reply:
Hmm, I think we are talking about two different things here:
1. The timezone of the `thing` and its fallbacks.
2. The timezone as it should be presented to the user.

For point 2 I am always using the currently configured time in `convertUnitToDateTimeType()`

```
// we provide the time in the format as configured in the openHAB settings
 ZonedDateTime zdt = adjustedInstant.atZone(timeZoneProvider.getTimeZone());
```

as its always requested from the `timeZoneProvider`.

For point 1. the information from the `timeZoneProvider` acts as a fallback only for the lifetime of the `Thinghandler`, or until someone sets the correct time on the `thing` (because then `dispose` and `initialize` will be called anyway.

So do you want me to change this, even if its just a fallback and even a warning is printed to the user to make him aware of that he should add the correct zone to his `Thing` configuration? I am fine with doing so, if you think its better. I just wanted to clarify things and see if I have understood you correctly.

Reply:
> I am fine with doing so, if you think its better. I just wanted to clarify things and see if I have understood you correctly.

Yep you have understood me correctly.

Reply:
Great :)

i have now changed it in a way that the warning about the not set timezone on the thing will only be printed once on the startup of the handler and if it is not set, I get it from the `timezoneProvider` at runtime for every call.